pandas/tests/io/pytables/store/test_read.py

```diff
@@ -0,0 +1,348 @@
+        # GH 24510
+        # legacy table with fixed format written in Python 2
+        with ensure_clean_store(
+            datapath("io", "data", "legacy_hdf", "legacy_table_fixed_py2.h5"), mode="r"
+        ) as store:
+            result = store.select("df")
+            expected = DataFrame(
+                [[1, 2, 3, "D"]],
+                columns=["A", "B", "C", "D"],
+                index=Index(["ABC"], name="INDEX_NAME"),
+            )
+            tm.assert_frame_equal(expected, result)
+
+    def test_legacy_table_fixed_format_read_datetime_py2(self, datapath, setup_path):
+        # GH 31750
+        # legacy table with fixed format and datetime64 column written in Python 2
+        with ensure_clean_store(
+            datapath("io", "data", "legacy_hdf", "legacy_table_fixed_datetime_py2.h5"),
+            mode="r",
+        ) as store:
+            result = store.select("df")
+            expected = DataFrame(
+                [[Timestamp("2020-02-06T18:00")]],
+                columns=["A"],
+                index=Index(["date"]),
+            )
+            tm.assert_frame_equal(expected, result)
+
+    def test_legacy_table_read_py2(self, datapath, setup_path):
+        # issue: 24925
+        # legacy table written in Python 2
+        with ensure_clean_store(
+            datapath("io", "data", "legacy_hdf", "legacy_table_py2.h5"), mode="r"
+        ) as store:
+            result = store.select("table")
+
+        expected = DataFrame({"a": ["a", "b"], "b": [2, 3]})
+        tm.assert_frame_equal(expected, result)
+
+    def test_read_hdf_open_store(self, setup_path):
+        # GH10330
+        # No check for non-string path_or-buf, and no test of open store
+        df = DataFrame(np.random.rand(4, 5), index=list("abcd"), columns=list("ABCDE"))
+        df.index.name = "letters"
+        df = df.set_index(keys="E", append=True)
+
+        with ensure_clean_path(setup_path) as path:
+            df.to_hdf(path, "df", mode="w")
+            direct = read_hdf(path, "df")
+            store = HDFStore(path, mode="r")
+            indirect = read_hdf(store, "df")
+            tm.assert_frame_equal(direct, indirect)
+            assert store.is_open
+            store.close()
+
+    def test_read_hdf_iterator(self, setup_path):
+        df = DataFrame(np.random.rand(4, 5), index=list("abcd"), columns=list("ABCDE"))
+        df.index.name = "letters"
+        df = df.set_index(keys="E", append=True)
+
+        with ensure_clean_path(setup_path) as path:
+            df.to_hdf(path, "df", mode="w", format="t")
+            direct = read_hdf(path, "df")
+            iterator = read_hdf(path, "df", iterator=True)
+            assert isinstance(iterator, TableIterator)
+            indirect = next(iterator.__iter__())
+            tm.assert_frame_equal(direct, indirect)
+            iterator.store.close()
+
+    def test_read_hdf_errors(self, setup_path):
```

Comment:
this is an errors test

Reply:
Yeah I wasn't super consistent about whether I left the errors with the tests for the method that raised the error, or in the separate errors section. I will look through to see if I can move more to the errors

Reply:
its no big deal, likely will have a followup pass anyhow for fixups.

Reply:
moved this and `test_read_hdf_generic_buffer_errors` to test_errors